hc_api/src/lib.rs

```diff
@@ -0,0 +1,78 @@
+/*!
```

Comment:
do we want to standardize on `/*!` or `//!`?

Reply:
well for me `//!` feels like a pain because I have to redo it if I want to change line breaks...  But maybe I should have tool to fix that?  Thoughts?

Reply:
awesome, let's go with `/*!` - i'll change the dna stuff to that for my next commit

Reply:
FYI the Rust API conventions states:
```
Only use inner doc comments //! to write crate and module-level documentation, nothing else. When using mod blocks, prefer /// outside of the block
```